torch_xla/csrc/xla_sharding_util.cpp

```diff
@@ -498,21 +504,9 @@ std::vector<XLATensor::ShardingSpecPtr> ShardingUtil::GetOutputSharding(
   }
 
   for (int i = 0; i < num_outputs; ++i) {
-    if (output_shardings[i].type()) {
-      // Tensor sharding annotation type is non-zero (sharded).
-      sharding_specs[i] = std::make_shared<XLATensor::ShardingSpec>(
-          output_shardings[i],
-          MakeShapeWithDeviceLayout((*output_shapes)[i],
-                                    static_cast<XlaDeviceType>(device.type())));
-    } else {
-      // Clear sharding if the output parameter is no longer sharded, this
-      // assumes that the output is implicitly replicated and wrapped inside
-      // PjRtShardedData.
-      sharding_specs[i] = std::make_shared<XLATensor::ShardingSpec>(
-          xla::HloSharding::Replicate().ToProto(),
-          MakeShapeWithDeviceLayout((*output_shapes)[i],
-                                    static_cast<XlaDeviceType>(device.type())));
-    }
+    sharding_specs[i] = std::make_shared<XLATensor::ShardingSpec>(
+        output_shardings[i],
+        MakeShapeWithDeviceLayout(output_shapes[i], XlaDeviceType::SPMD));
```

Comment:
I guess we don't mark output as explicit replicated anymore and that's OK?

Reply:
We want it to be explicitly replicated and it is. Ouput shardings are all replicated by default.